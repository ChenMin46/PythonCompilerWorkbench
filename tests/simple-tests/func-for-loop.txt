def foo(a, b, c=3):
    d = a + b * c - bar()

    for i in range(10):
        print i, d

    return c, d
