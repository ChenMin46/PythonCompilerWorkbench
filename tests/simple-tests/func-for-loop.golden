def foo(a, b, c=3):
    d = a + b * c - bar()

    for i in range(10):
        print i, d

    return c, d
===
Module
  body: [
    FunctionDef >> 'def foo(a, b, c=3)'
      name: 'foo'
      args: arguments
        args: [
          Name >> 'a'
            id: 'a'
            ctx: Param
          Name >> 'b'
            id: 'b'
            ctx: Param
          Name >> 'c'
            id: 'c'
            ctx: Param
        ]
        vararg: None
        kwarg: None
        defaults: [
          Num >> '3'
            n: 3
        ]
      body: [
        Assign >> 'd = a + b * c - bar()'
          targets: [
            Name >> 'd'
              id: 'd'
              ctx: Store
          ]
          value: BinOp >> 'a + b * c - bar()'
            left: BinOp >> 'a + b * c'
              left: Name >> 'a'
                id: 'a'
                ctx: Load
              op: Add
              right: BinOp >> 'b * c'
                left: Name >> 'b'
                  id: 'b'
                  ctx: Load
                op: Mult
                right: Name >> 'c'
                  id: 'c'
                  ctx: Load
            op: Sub
            right: Call >> 'bar()'
              func: Name >> 'bar'
                id: 'bar'
                ctx: Load
              args: [
              ]
              keywords: [
              ]
              starargs: None
              kwargs: None
        For >> 'for i in range(10)'
          target: Name >> 'i'
            id: 'i'
            ctx: Store
          iter: Call >> 'range(10)'
            func: Name >> 'range'
              id: 'range'
              ctx: Load
            args: [
              Num >> '10'
                n: 10
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
          body: [
            Print >> 'print i, d'
              dest: None
              values: [
                Name >> 'i'
                  id: 'i'
                  ctx: Load
                Name >> 'd'
                  id: 'd'
                  ctx: Load
              ]
              nl: True
          ]
          orelse: [
          ]
        Return >> 'return c, d'
          value: Tuple >> 'c, d'
            elts: [
              Name >> 'c'
                id: 'c'
                ctx: Load
              Name >> 'd'
                id: 'd'
                ctx: Load
            ]
            ctx: Load
      ]
      decorator_list: [
      ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            "def foo(", 
            {
              "contents": [
                {
                  "contents": [
                    "a"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "b"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                "=", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "arguments"
            }, 
            ")"
          ], 
          "id": "id_0", 
          "name": "FunctionDef"
        }, 
        ":\n    ", 
        {
          "contents": [
            {
              "contents": [
                "d"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " = ", 
            {
              "contents": [
                {
                  "contents": [
                    {
                      "contents": [
                        "a"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " + ", 
                    {
                      "contents": [
                        {
                          "contents": [
                            "b"
                          ], 
                          "id": "id_0", 
                          "name": "Name"
                        }, 
                        " * ", 
                        {
                          "contents": [
                            "c"
                          ], 
                          "id": "id_0", 
                          "name": "Name"
                        }
                      ], 
                      "id": "id_0", 
                      "name": "BinOp"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }, 
                " - ", 
                {
                  "contents": [
                    {
                      "contents": [
                        "bar"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    "()"
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n\n    ", 
        {
          "contents": [
            "for ", 
            {
              "contents": [
                "i"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " in ", 
            {
              "contents": [
                {
                  "contents": [
                    "range"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                "(", 
                {
                  "contents": [
                    "10"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        ":\n        ", 
        {
          "contents": [
            "print ", 
            {
              "contents": [
                "i"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            ", ", 
            {
              "contents": [
                "d"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n    ", 
        {
          "contents": [
            "return ", 
            {
              "contents": [
                {
                  "contents": [
                    "c"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                ", ", 
                {
                  "contents": [
                    "d"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }
          ], 
          "id": "id_0", 
          "name": "Return"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
